### ISSUE TITLE
Buttons on the WebUI

### ISSUE THREAD

**Issue by 0xFDEC0DED** [OpenPoGo/OpenPoGoUI#28]
The Buttons on the WebUI for displaying Pokemons, Eggs and Items unfortunately do not work for me. Also most of the time the map does not display Pokemon, Pokestops and Arenas.

**Comment by niicojs** [comment]
Could you give us more info? Like browser, url you use to access the gui?
Are you up to date on the bot?

Also could you paste here console log (from the bot) and browser log (from browser console)?

**Comment by 0xFDEC0DED** [comment]
Im using Google Chrome. Im connecting to 127.0.0.1:8000. uBlock Origin is disabled.. 

**Console Log when connecting to the UI:**

(5344) accepted ('127.0.0.1', 53729)
[2016-09-04 13:25:49] [UI] Web client connected
(5344) accepted ('127.0.0.1', 53733)
(5344) accepted ('127.0.0.1', 53734)
(5344) accepted ('127.0.0.1', 53735)
(5344) accepted ('127.0.0.1', 53736)

**Browser Log when im connecting:**

Load config from storage
bundle.js:1 Connecting to http://localhost:8000
bundle.js:1 Connected to Bot
bundle.js:1 Bot Ready.
bundle.js:1 Objectcoordinates: Array[3]level: _censored_player: Objectstorage: Objectusername: "_censored_"**proto**: Object

**Browser Log when im Pressing the "ShowPokemon" Button:**
.. I don't get any Browser-Console Log ..

I **do** get to Display the Pokemon-List when I am continuously pushing the Button (for showing the pkm list) on page reload (F5).

**Comment by niicojs** [comment]
Hi. This shouldn't be a problem, but you don't have to run the UI locally. Just go to http://openpogoui.nicontoso.eu/, it will work the same.

Also, I'm trouble understanding all those "accepted" log. Do you use virtual env to run the bot?

**Comment by niicojs** [comment]
Also, do you see a log on the bot console when you click on a UI button?

**Comment by 0xFDEC0DED** [comment]
My local service on 8000 redirects me to http://openpogoui.nicontoso.eu/ 

The "accepted" is displayed in my console while i am connecting to my local service.

I don't know whether I am using a virtual env. I have no experince with python.
Maybe you can help me figure it out.

Edit: I just got my hands on some information about the virtual env.. I guess i am not using one..

No there is no change in the Console when i click on a Button in the UI.

**Comment by niicojs** [comment]
Could you try using virtualenv? I guess there is something that conflicts with one of the prerequisites.

**Comment by 0xFDEC0DED** [comment]
ok thanks i will try this out later..

**Comment by unarmedcow** [comment]
I had the same problem.  Running it in virtualenv makes it work for me.

Without virtualenv, it appears that OpenPoGoUI wasn't receiving events emitted from OpenPoGoBot.  Enabling debugging for the socket plugin for OpenPoGoBot allowed me to see that OpenPoGoBot was receiving events from OpenPoGoUI, but when OpenPoGoBot responded with an event, the event never appeared in the console of the browser in OpenPoGoUI.

Looking into it, this might be the problem:
https://github.com/miguelgrinberg/Flask-SocketIO/issues/253

Basically Flask-SocketIO (the library OpenPoGoBot uses for socket-io communications) doesn't work with npm's socket.io-client because of an alleged bug in socket.io-client's code.  The Flask-SocketIO author has submitted a pull request to fix it here: https://github.com/socketio/engine.io-client/pull/483/files
